gnosis/safe-react
# Use single ga

## What it solves
Resolves #2703

## How this PR fixes it
All GA tracking is now done under one ID (previously only used for mainnet). All other `REACT_APP_GOOGLE_ANALYTICS` env vars have been removed from the deploy scripts accordingly. @katspaugh, maybe we should update our secrets in the future to reflect this change?

`ReactGA` uses GA's own `ga()` method to track events because we now send a custom `chain` variable, which is the label of the current network.

The removal of GA `COOKIES_LIST` has also been simplified for cookie removal.

Note: this includes the changes [here](https://github.com/gnosis/safe-react/issues/2542), logging in the dev environment.

## How to test it
- Load the app in the `dev` environment and check that GA is logging locally.
- Load the app in the `staging` environment and check that GA is initialised.
  - Switch to non-mainnet networks and ensure that tracking is registered in GA (with custom `chain` var).
  - Uncheck the GA tracking cookie optiona and check that the cookie is deleted/tracking does not occur.

## Screenshots
![image](https://user-images.githubusercontent.com/20442784/135238871-a284dd53-7a00-4ed3-8759-4dce56b87008.png)
